Tests/test_function.py

```diff
@@ -287,6 +287,65 @@ def __init__(self, group=None, target=None):
 
         foo.__init__(a, target='baz')
 
+    def test_kwargs3(self):
+        """verify keyword args with keys of subtypes of str"""
+
+        class mystr(str): pass
+
+        class mymapping:
```

Comment:
Could wrap a dict instead:
```Python
class mymapping:
    def __init__(self, d):
        self.d = d
    def keys(self):
        return self.d.keys()
    def __len__(self):
        return self.d.__len__()
    def __getitem__(self, x):
        return self.d.__getitem__(x)
```
although maybe not worth it...

Reply:
Yes, good idea, it's cleaner and maybe copy/paste-reusable somewhere else.